src/lib/preview-websocket.ts

```
@@ -0,0 +1,382 @@
+import { PreviewCache } from './preview-data';
+
+export interface MockSocketEvent {
+    event: string;
+    data: any;
+    timestamp: string;
+}
+
+export class PreviewWebSocketService {
+    private static instance: PreviewWebSocketService;
+    private cache: PreviewCache;
+    private eventHandlers: Map<string, Function[]> = new Map();
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)